homeassistant/components/eheimdigital/entity.py

```diff
@@ -0,0 +1,55 @@
+"""Base entity for EHEIM Digital."""
+
+from abc import abstractmethod
+from typing import TYPE_CHECKING
+
+from eheimdigital.device import EheimDigitalDevice
+
+from homeassistant.const import CONF_HOST
+from homeassistant.core import callback
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC, DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import EheimDigitalUpdateCoordinator
+
+
+class EheimDigitalEntity[_DeviceT: EheimDigitalDevice](
+    CoordinatorEntity[EheimDigitalUpdateCoordinator]
+):
+    """Represent a EHEIM Digital entity."""
+
+    _attr_has_entity_name = True
+    _device: _DeviceT
+    _device_address: str
+
+    def __init__(
+        self, coordinator: EheimDigitalUpdateCoordinator, device: _DeviceT
+    ) -> None:
+        """Initialize a EHEIM Digital entity."""
+        super().__init__(coordinator)
+        if TYPE_CHECKING:
+            # At this point at least one device is found and so there is always a main device set
+            assert isinstance(coordinator.hub.main, EheimDigitalDevice)
```

Comment:
I am wondering, what else can it be?

Reply:
For type checkers it can still be `None` if the main device wouldn't get set